### PR TITLE
naming : normalize the name of callback-related identifiers

### PR DESCRIPTION
**Pull request by ggerganov**
Renames identifiers into the following convention:

```c
foo_callback cb_foo;
void *       cb_foo_ctx;
```

Or when the callback context is unambiguous, simply to:

```c
void foo_set(foo_callback cb, void * cb_ctx);
```

Request for comments: is this change worth it?

### CODE REVIEW

File: include/llama.h [hunk 1]
```diff
@@ -331,25 +329,24 @@ extern "C" {
...
-        void *              abort_callback_data;
+        ggml_abort_callback cb_abort;
+        void *              cb_abort_ctx;
+
+        // Keep the booleans together and at the end of the struct to avoid misalignment during copy-by-value.
+        bool logits_all;  // the llama_decode() call computes all logits, not just the last one (DEPRECATED - set llama_batch.logits instead)
+        bool embeddings;  // if true, extract embeddings (together with logits)
+        bool offload_kqv; // offload the KQV ops (including the KV cache) to GPU
+        bool flash_attn;  // enable flash attention [EXPERIMENTAL]
+        bool no_perf;     // disable performance timings
```
Comment (ggerganov) [inline include/llama.h:349]: Breaks `libllama` API because of parameter reordering and renaming
